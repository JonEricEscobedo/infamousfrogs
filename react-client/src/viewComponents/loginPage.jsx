import React from 'react';
import $ from 'jquery';


class loginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    }

  this.passwordChange = this.passwordChange.bind(this);
  this.userChange = this.userChange.bind(this);
  this.formSubmit = this.formSubmit.bind(this);

}

passwordChange(e) {
  e.preventDefault();
  this.setState({password: e.target.value})
}

userChange(e) {
  e.preventDefault();
  this.setState({username: e.target.value})
}

formSubmit(e) {
  e.preventDefault();
  var obj = {
    username: this.state.username,
    password: this.state.password
  }
  var self = this;
  $.ajax({
    type: 'POST',
    url: '/login',
    contentType: 'application/json',
    data: JSON.stringify(obj),
    dataType: 'text',
    success: (data) => {
      data = JSON.parse(data);
      if (data.id) {
        self.props.history.push('/')
      }
      else if (data.userdoesnotexist) {
        alert("User does not exist! Try again!")
      }
      else if (data.incorrectpassword) {
        alert("Password is Incorrect! Try again!")
      }
    }
  });

}

render() {
 return (
  <div className="container">
      <div className="row">
  		<div className="span12">
  			<div className="form-horizontal">
  			  <fieldset>
  			    <div id="legend">
  			      <legend className="">Login</legend>
  			    </div>
  			    <div className="control-group">
  			      <label className="control-label"  htmlFor="username">Username</label>
  			      <div className="controls">
  			        <input onChange = {this.userChange} type="text" id="username" name="username" placeholder="" className="input-xlarge" />
  			      </div>
  			    </div>
  			    <div className="control-group">
  			      <label className="control-label" htmlFor="password">Password</label>
  			      <div className="controls">
  			        <input onChange = {this.passwordChange} type="password" id="password" name="password" placeholder="" className="input-xlarge" />
  			      </div>
  			    </div>
  			    <div className="control-group">
  			      <div className="controls">
  			        <button onClick = {this.formSubmit} className="btn btn-success">Login</button>
  			      </div>
  			    </div>
  			  </fieldset>
  			</div>
  		</div>
  	</div>
  </div>

)
}
}

export default loginPage
