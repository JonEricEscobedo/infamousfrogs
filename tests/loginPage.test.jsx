import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import loginPage from '../react-client/src/viewComponents/loginPage.jsx';

test('loginPage component should render as expected', () => {
  const component = shallow(<loginPage />)
  const tree = toJson(component)

  expect(tree).toMatchSnapshot()
})