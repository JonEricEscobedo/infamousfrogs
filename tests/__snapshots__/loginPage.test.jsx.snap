// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginPage component should render as expected 1`] = `<loginPage />`;
